java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -684,9 +686,14 @@ private int getTypeBufferValueCapacity() {
 
     @Override
     public int hashCode(int index) {
-      return getVector(index).hashCode(index);
+      return hashCode(index, null);
     }
 
+    @Override
+    public int hashCode(int index, ArrowBufHasher hasher) {
+    return getVector(index).hashCode(index, hasher);
```

Comment:
@tianchen92 the formatting seems wrong here

Reply:
Thanks, fixed.